File: hwk5/cens_fit.py
## Imports
import numpy as np
import cvxpy as cp

## Fitting with censored data
X_file = "~/development/cvxopt/hwk5/X.csv"
y_file = "~/development/cvxopt/hwk5/y.csv"
c_true_file = "~/development/cvxopt/hwk5/c_true.csv"

n = 20
M = 25
K = 100
D = -1.7712

X = np.loadtxt(open(X_file, "rb"), delimiter=",")
y = np.loadtxt(open(y_file, "rb"), delimiter=",")
c_true = np.loadtxt(open(c_true_file, "rb"), delimiter=",")

z = cp.Variable(K-M)
c = cp.Variable(n)
ss_low = cp.sum_squares(y - c.T @ X[:,0:M])
ss_upp = cp.sum_squares(z - c.T @ X[:,M:K])
ss_all = ss_low + ss_upp

p = cp.Problem(cp.Minimize(ss_all), [z >= D])
p.solve()

r_opt = cp.norm(c_true - c, 2) / cp.norm(c_true)

File: hwk5/cens_fit.R
library(data.table)
library(ggplot2)

n <- 20
M <- 25
K <- 100
D <- -1.7712

y <- fread("~/development/cvxopt/hwk5/y.csv")$V1
X <- t(as.matrix(fread("~/development/cvxopt/hwk5/X.csv")))
c_true <- fread("~/development/cvxopt/hwk5/c_true.csv")$V1

data <- as.data.frame(cbind(X[1:25,], y))
c_ls <- lm("y ~ . - 1", data)$coefficients
r_ls <- sqrt(sum((c_true - c_ls)^2)) / sqrt(sum(c_true^2))
